### ISSUE TITLE
Egress does not ACTIVE, sending plenty of PLI requests to publisher.

### ISSUE THREAD

**Issue by namnt1810** [livekit/livekit#2146]
**Describe the bug**
We are using self-hosted Livekit server deployed on kubernetes clusters. We have problem when using Egress to record a participant's stream. Egress sometimes successfully record the video but most of the time it fails.

**Server**
- Version: 1.4.5
- Egress versions: 1.7.9
- Environment: EKS

<details>
  <summary>LiveKit server config</summary>  

```yaml
prometheus_port: 6789
  log_level: debug
  rtc:
    use_external_ip: false
    turn_servers:
      - host: ...
        port: 3478
        # tls, tcp, or udp
        protocol: udp
        username: ...
        credential: ...
  redis:
    address: :6379
    password: ...
  keys:
    ...
  turn:
    enabled: false
    # domain: <turn-address>
    # tls_port: 3478
    # secretName: <turn-tls-secret>
  egress:
    # Whether to use the PSRPC enabled RPC implementation. This requires livekit egress version >=1.5.4
    # The legacy, non PSRPC RPC implementation will be removed eventually
    use_psrpc: false
  webhook:
    # the API key to use in order to sign the message
    # this must match one of the keys LiveKit is configured with  
    api_key: $LIVEKIT_ID
    urls:
      - ...
  room:
    enabled_codecs:
      - mime: video/vp8
      - mime: audio/opus
      - mime: audio/red
```
</details>

<details>  
  <summary>Egress service config</summary>  

  ```yaml
  # required fields
  api_key: ...
  api_secret: ...
  ws_url: wss://...
  redis:
    address: :6379
    password: ...
  # optional fields
  log_level: debug
  health_port: 8080
  prometheus_port: 9090
  insecure: true
  cpu_cost: # optionally override cpu cost estimation, used when accepting or denying requests
    room_composite_cpu_cost: 2.5
    web_cpu_cost: 2.5
    track_composite_cpu_cost: 1
    track_cpu_cost: 0.5
  # file upload config - only one of the following. Can be overridden per-request
  gcp:
    bucket: ...
  ```
</details>
  
**Client**
- SDK: iOS
- Version: 1.0.13

<details>  
  <summary>iOS client config</summary>  

  ```swift
  let room = Room(delegate: self)

  let rtcConfig = RTCConfiguration()
  rtcConfig.iceTransportPolicy = .relay
  let options = ConnectOptions(rtcConfiguration: rtcConfig)
  
  var simulcastLayers = [VideoParameters]()
  var encoding = VideoParameters.presetH720_169.encoding
  var dimension = VideoParameters.presetH720_169.dimensions
  if let config = self.config {
      encoding = VideoEncoding(maxBitrate: config.bitrate, maxFps: 30)

      let simulcastKeys = config.simulcastLayers
      for key in simulcastKeys {
          let layer: VideoParameters
          switch (key) {
          case "h180":
              layer = .init(dimensions: .h180_169, encoding: encoding)
              break;
          case "h360":
              layer = .init(dimensions: .h360_169, encoding: encoding)
              break;
          case "h1080":
              layer = .init(dimensions: .h1080_169, encoding: encoding)
              break;
          default:
              layer = .init(dimensions: .h720_169, encoding: encoding)
              break
          }
          simulcastLayers.append(layer)
      }
      let highest = simulcastLayers.sorted(by: { first, second in
          return first.dimensions.height < second.dimensions.height
      }).first

      if let highest = highest {
          dimension = highest.dimensions
      }
  }
  
  let roomOptions = RoomOptions(
            defaultCameraCaptureOptions: .init(position: cameraPosition,
                                               dimensions: dimension
                                              ),
            defaultVideoPublishOptions: .init(encoding: encoding,
                                              simulcast: simulcastLayers.count > 1,
                                              simulcastLayers: simulcastLayers
                                             )
        )

  room.connect(url, token, connectOptions: options, roomOptions: roomOptions)
  ```

</details>

**To Reproduce**  
1. Participant A join room 1.
1. Backend server wait for track publish events of Participant A to request start an Egress.
1. Starting Egress by calling StartTrackCompositeEgress with Participant A video and audio track id, output type is segments.
1. Backend server stop the Egress. The Egress status is `ABORTED`.

**Expected behavior**
Egress records successfully.

**Screenshots**

**Additional context**
We did have the server logs but it's too long so I cannot paste it here.  
From the logs, we saw a message indicates `videolayerselector/simulcast` tried to upgrade the layer but resulting rolling back afterwards.

```
08:08:48.657Z	DEBUG	livekit.sub	videolayerselector/simulcast.go:51	upgrading layer	{"room": "5dabd19e-c353-4784-9738-92fa06a93605", "roomID": "RM_i4CDBipMZbkw", "participant": "EG_C8rfKr5T9c7W", "pID": "PA_YKuitaEojwMS", "remote": false, "trackID": "TR_VC7VYYX8JwH7AF", "relayed": false, "previous": "VideoLayer{s: -1, t: -1}", "current": "VideoLayer{s: 0, t: 0}", "previousTarget": "VideoLayer{s: 0, t: 3}", "target": "VideoLayer{s: 0, t: 3}", "max": "VideoLayer{s: 0, t: 3}", "layer": 0, "req": 0, "maxSeen": "VideoLayer{s: 0, t: -1}", "feed": 2670240106}
08:08:48.657Z	DEBUG	livekit.sub	videolayerselector/base.go:129	rolling back	{"room": "5dabd19e-c353-4784-9738-92fa06a93605", "roomID": "RM_i4CDBipMZbkw", "participant": "EG_C8rfKr5T9c7W", "pID": "PA_YKuitaEojwMS", "remote": false, "trackID": "TR_VC7VYYX8JwH7AF", "relayed": false, "previous": "VideoLayer{s: -1, t: -1}", "current": "VideoLayer{s: 0, t: 0}", "previousTarget": "VideoLayer{s: 0, t: 3}", "target": "VideoLayer{s: 0, t: 3}", "max": "VideoLayer{s: 0, t: 3}", "req": 0, "maxSeen": "VideoLayer{s: 0, t: -1}"}
```
  
Then there is a series of `sending PLI for layer lock` message.
```
08:08:48.660Z	DEBUG	livekit.sub	sfu/downtrack.go:603	sending PLI for layer lock	{"room": "5dabd19e-c353-4784-9738-92fa06a93605", "roomID": "RM_i4CDBipMZbkw", "participant": "EG_C8rfKr5T9c7W", "pID": "PA_YKuitaEojwMS", "remote": false, "trackID": "TR_VC7VYYX8JwH7AF", "relayed": false, "generation": 2, "layer": 0}
```
  
What should we do in this case to make the Egress start properly.

Thanks.

**Comment by davidzhao** [comment]
How are you publishing the track? Is it end to end encrypted?

**Comment by namnt1810** [comment]
@davidzhao we are publishing tracks after the connect sequence is successful, like this:
```swift
room.connect(url, token, connectOptions: options, roomOptions: roomOptions)
            .then { r in
                r.localParticipant?.setMicrophone(enabled: true)
                r.localParticipant?.setCamera(enabled: true)
                    .then { [weak self] publication in
                        guard let weakSelf = self,
                              let publication = publication else { return }
                        weakSelf.registerFrameProcessor(publication: publication)
                    }
                UIApplication.shared.isIdleTimerDisabled = true
            }.catch { error in
                logger.log(level: .error, "connect error:\(error)")
            }
```
And we don't use E2E encryption, our client sdk version is 1.0.13, which does not include the E2EE feature.

**Comment by davidzhao** [comment]
Can you give livekit-server 1.5.0 a try? It contains improvements with TrackComposite Egress.

**Comment by namnt1810** [comment]
Let us try that version.

Thanks.

P/s: we tried the same swift client with your [Cloud SaaS server](https://cloud.livekit.io/login) and the Egress recording works properly.

**Comment by giovankabisano** [comment]
Hi @namnt1810. Just wondering because I'm gonna use this service very soon. 
Is your problem still persist after using livekit-server 1.5.0?

**Comment by namnt1810** [comment]
Hi @giovankabisano, we have updated our Livekit version to 1.5.0 and are observing the issue.
So far, everything is good.
We will be observing in the next week and will update the status later then.

**Comment by davidzhao** [comment]
closing since it appears to be resolved with 1.5.0, please reopen if issue persists.
